extensions/common/context.cc

```diff
@@ -449,7 +454,25 @@ void populateTCPRequestInfo(bool outbound, RequestInfo* request_info) {
   // host_header_fallback is for HTTP/gRPC only.
   populateRequestInfo(outbound, false, request_info);
 
-  request_info->request_protocol = kProtocolTCP;
+  uint64_t response_flags = 0;
+  if (getValue({"response", "flags"}, &response_flags)) {
+    request_info->response_flag = parseResponseFlag(response_flags);
+  }
+
+  request_info->request_protocol = Protocol::TCP;
```

Comment:
Don't need this? as we set it in onNewConnection and same for HTTP and Grpc.. they are getting set in onRequestHeaders?

Reply:
It's not set in stackdriver. I suppose we could move this line into stackdriver plugin itself.